runtime/src/stash_measurement.rs

```diff
@@ -4,20 +4,27 @@ use crate::{dpe_crypto::DpeCrypto, CptraDpeTypes, DpePlatform, Drivers};
 use caliptra_common::mailbox_api::{
     MailboxResp, MailboxRespHeader, StashMeasurementReq, StashMeasurementResp,
 };
-use caliptra_drivers::{CaliptraError, CaliptraResult};
+use caliptra_drivers::{pcr_log::PCR_ID_STASH_MEASUREMENT, CaliptraError, CaliptraResult};
 use crypto::{AlgLen, Crypto};
 use dpe::{
     commands::{CommandExecution, DeriveChildCmd, DeriveChildFlags},
     context::ContextHandle,
     dpe_instance::DpeEnv,
     response::DpeErrorCode,
 };
-use zerocopy::FromBytes;
+use zerocopy::{AsBytes, FromBytes};
 
 pub struct StashMeasurementCmd;
 impl StashMeasurementCmd {
     pub(crate) fn execute(drivers: &mut Drivers, cmd_args: &[u8]) -> CaliptraResult<MailboxResp> {
         if let Some(cmd) = StashMeasurementReq::read_from(cmd_args) {
+            // Extend the measurement into PCR31
+            drivers.pcr_bank.extend_pcr(
+                PCR_ID_STASH_MEASUREMENT,
+                &mut drivers.sha384,
+                cmd.measurement.as_bytes(),
+            )?;
```

Comment:
Ideally we would wait to extend the PCR until we're sure the rest of the command will succeed. This is still a little annoying because if pcr_extend fails, we also probably want to rollback the DeriveChild call which might be difficult to do in practice. But the PCR extend is for sure irreversible, so we should do it last.

Reply:
The issue then is that we get borrow checker issues since we "cannot borrow `drivers.sha384` as mutable more than once at a time". I will look into refactoring this logic a bit

Reply:
Fixed by limiting the scope of the first borrow of drivers.sha384.